lib/evaluate.js

```diff
@@ -272,8 +273,7 @@ export const runFraudDetection = async (roundIndex, measurements, sparkRoundDeta
   for (const m of measurements) {
     if (m.fraudAssessment) continue
 
-    // TODO: include Filecoin Deal id
-    const key = `${m.inet_group}::${m.cid}`
+    const key = `${m.inet_group}::${m.cid}::${m.minerId}`
```

Comment:
FYI: I first updated a test to fail with an expected error and only then made this change.